packages/provider-utils/src/test/unified-test-server.ts

```diff
@@ -104,6 +122,7 @@ export function createTestServer<URLS extends { [url: string]: UrlHandler }>(
 ): {
   urls: FullHandlers<URLS>;
   calls: TestServerCall[];
+  addSseEvent: (url: string, data: any, event?: string, id?: string) => void;
```

Comment:
wondering about this because it is sse specific. there is another pattern i used, see e.g. https://github.com/vercel/ai/blob/main/packages/react/src/use-chat.ui.test.tsx#L880

wondering if that would be an alternative (maybe it's the same thing and `addServerSentEvent` or `addSSE` (rename?) is more elegant

Reply:
also do we want to be SSE specific here? this could just a generic stream and you could attach SSE formatted chunks

Reply:
my approach is missing a sort of `streamController` like your approach is 👀 hoping to discuss in call tomo!